waspc/src/Wasp/Psl/Parser/Common.hs

```diff
@@ -13,66 +13,105 @@ module Wasp.Psl.Parser.Common
     float,
     integer,
     SourceCode,
+    Parser,
   )
 where
 
-import Text.Parsec
-  ( alphaNum,
-    char,
-    letter,
+import Data.Functor (void)
+import Data.Void (Void)
+import Text.Megaparsec
+  ( Parsec,
+    between,
+    empty,
+    many,
+    manyTill,
+    notFollowedBy,
+    sepBy,
+    sepBy1,
+    takeWhileP,
+    try,
+    (<?>),
     (<|>),
   )
-import Text.Parsec.Language (emptyDef)
-import Text.Parsec.String (Parser)
-import qualified Text.Parsec.Token as T
+import qualified Text.Megaparsec.Char as C
+import qualified Text.Megaparsec.Char.Lexer as L
 
-type SourceCode = String
+type Parser = Parsec Void SourceCode
 
-whiteSpace :: Parser ()
-whiteSpace = T.whiteSpace lexer
+type SourceCode = String
 
 reserved :: String -> Parser ()
-reserved = T.reserved lexer
+reserved name =
+  lexeme $
+    try $
+      do
+        _ <- C.string name
+        notFollowedBy identLetter <?> ("end of " ++ show name)
 
 identifier :: Parser String
-identifier = T.identifier lexer
+identifier = lexeme $ try ident
+  where
+    ident =
+      do
+        c <- identStart
+        cs <- many identLetter
+        return (c : cs)
+        <?> "identifier"
```

Comment:
Why not like this (simpler)?

```suggestion
identifier =
  lexeme $ try $ do
    c <- identStart
    cs <- many identLetter
    return (c : cs) <?> "identifier"
```

Reply:
I also put that `<?> "identifier"` in same line with return, but maybe I don't know what that does hm. Why did you put it in the last line alone?
EDIT: Ok I read that is a way to label stuff to get nicer error messages. Sounds nicer to do it as I did in suggestion above?

Reply:
Nice! I would maybe inline the content of `ident`, but `identHead` and `identTail` are nice.